Makefile

```diff
@@ -27,31 +27,37 @@
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ###############################################################################
 
+.PHONY: all
 all: test
 
 include cocotb/share/makefiles/Makefile.inc
 
+.PHONY: clean
 clean:
 	-@rm -rf $(BUILD_DIR)
 	-@find . -name "obj" | xargs rm -rf
 	-@find . -name "*.pyc" | xargs rm -rf
 	-@find . -name "*results.xml" | xargs rm -rf
-	$(MAKE) -C examples clean
-	$(MAKE) -C tests clean
+	-$(MAKE) -C examples clean
+	-$(MAKE) -C tests clean
```

Comment:
While safe, this doesn't seem particularly useful - don't the submake targets ignore all their errors anyway?

Reply:
Not all apparently, I just checked. `make clean` fails halfway through because `target 'clean' not found`.

Reply:
We should add a clean target to that test then

Reply:
After looking through the makefiles, only `test_array` and `test_multidimensional_array` have issues and only when using Icarus. The ifdefs are missing `clean` targets. Should be an easy fix.

Reply:
I have done this now.